src/components/VirtualList/Config.ts

```diff
@@ -111,7 +136,7 @@ export const IstioTypes = {
 
 export type Resource = {
   name: string;
-  columns: ResourceType[];
+  columns: ResourceType<any>[];
```

Comment:
Note: I tried, but aborted propagation of generic as `Resource<R>` because it's generating too many changes and in some cases leading to very noisy definitions.

Reply:
Would the code benefit from using `unknown` instead of `any`? 
You could use typeguards (or a explicit cast) when you want to access the data.
This way we ensure that you have to be explicit on what you expect it to be when you access the data.

Reply:
thanks i'll try that tomorrow

Reply:
@josejulio I couldn't do it. Using just `ResourceType<unknown>` is not possible as ResourceType needs type that extends `TResource` ; which is ok with `any` but not with `unknown`.

Reply:
Ok, thanks for trying!